rootfs/api/models/build.py

```diff
@@ -52,8 +52,8 @@ def version(self):
         return 'git-{}'.format(self.sha) if self.source_based else 'latest'
 
     def create(self, user, *args, **kwargs):
-        latest_release = self.app.release_set.latest()
-
+        latest_release = self.app.release_set.filter(failed=False).latest()
```

Comment:
How would we re-deploy a failed release? There is no way other than to "roll with the changes", so to speak, and as far as I can see there's no code available in this PR to undo a failed deploy.